src/lib/helpers/jsToAqua.ts

```diff
@@ -24,14 +26,29 @@ import { capitalize } from "./capitilize.js";
 import { cleanAquaName, validateAquaName } from "./downloadFile.js";
 import { stringifyUnknown } from "./jsonStringify.js";
 
+/**
+ * In js object, json or yaml when you want to represent optional value and still generate a type for it you can use this syntax:
+ * { $$optional: <someValue>, $$isNil: <boolean> }
+ */
 const OPTIONAL = "$$optional";
+/**
+ * When `$$isNil == true`, the value is represented as `nil` in aqua but it will still have type inferred from `$$optional` value.
+ */
 const IS_NIL = "$$isNil";
 
+type EmptyObject = Record<string, never>;
+
+/**
+ * Empty object and empty array are both considered nil in aqua
```

Comment:
Are they? I don't think `{} == nil` will yield `true` in Aqua 🤔

Reply:
I fixed the description so it's more clear. There is just no {} in aqua at all